pkg/util/machines.go

```diff
@@ -140,12 +142,20 @@ func IsControlPlaneMachine(machine metav1.Object) bool {
 func GetMachineMetadata(hostname string, machine infrav1.VSphereVM, networkStatus ...infrav1.NetworkStatus) ([]byte, error) {
 	// Create a copy of the devices and add their MAC addresses from a network status.
 	devices := make([]infrav1.NetworkDeviceSpec, len(machine.Spec.Network.Devices))
+
+	// checking that the vSphere virtual machine's network status, if any
+	// match the number of network devices for the VSphereVM
+	if statuses := len(networkStatus); statuses > 0 {
+		if statuses != len(machine.Spec.Network.Devices) {
+			machine.Status.FailureMessage = pointer.StringPtr("Incorrect number of network devices. A network device was removed from the VM.")
+			machine.Status.FailureReason = capierrors.MachineStatusErrorPtr(capierrors.InvalidConfigurationMachineError)
```

Comment:
we shouldn't be setting `FailureMessage` and `FailureReason` on the machine directly. Those should be set at the `VSphereVM` level (our controllers will bubble it up to the `VSphereMachine` and CAPI will read from it)

Reply:
changed machine -> vsphereVM

Reply:
thanks